### PR TITLE
Secret base

### PR DESCRIPTION
**Pull request by PikalaxALT**

### CODE REVIEW

File: src/secret_base.c [hunk 1]
```diff
@@ -0,0 +1,2272 @@
...
+            {
+                data[1] = 1;
+            }
+            break;
+    }
+}
+
+void sub_80EA828(u8 sbId, struct SecretBaseRecord *base, u32 version, u32 language)
+{
+    int strlen;
```
Comment (Diegoisawesome) [inline src/secret_base.c]: strlen->stringLength

File: src/secret_base.c [hunk 2]
```diff
@@ -0,0 +1,2272 @@
...
+        {
+            return i;
+        }
+    }
+    return 0;
+}
+
+u8 sub_80EAA64(struct SecretBaseRecord *base, u32 version, u32 language)
+{
+    s16 sbId;
```
Comment (Diegoisawesome) [inline src/secret_base.c]: sbId->secretBaseID

File: src/secret_base.c [hunk 3]
```diff
@@ -0,0 +1,2272 @@
...
+    if (VarGet(VAR_0x40F0) && gUnknown_0203A01D == TRUE && !CurrentMapIsSecretBase())
+    {
+        VarSet(VAR_0x40F0, FALSE);
+        gUnknown_0203A01D = FALSE;
+        sub_80EEA70();
+        VarSet(VAR_0x40EC, 0);
+        VarSet(VAR_0x40ED, 0);
+        VarSet(VAR_0x40EE, 0);
+        VarSet(VAR_0x40EF, 0);
+        VarSet(VAR_0x40F0, FALSE);
```
Comment (Diegoisawesome) [inline src/secret_base.c]: Why are some of these set to 0 and some set to FALSE? This should be consistent.

Comment (PikalaxALT) [reply src/secret_base.c]: I'll double-check, but it's intended to reflect the difference between whether the var is ised as a u16 or as a bool16.

File: src/secret_base.c [hunk 4]
```diff
@@ -0,0 +1,2272 @@
...
+void sub_80E933C(void)
+{
+    u8 *roomdecor;
+    u8 *roomdecorpos;
+    u8 decidx;
+    u8 objid;
+    u8 metatile;
+    u8 category;
+    u8 permission;
+    u8 ndecor;
```
[the remark this answers is not in the document]
Comment (PikalaxALT) [reply src/secret_base.c]: ndecor, sure; but the rest are fine as is

[the remark this answers is not in the document]
Comment (Diegoisawesome) [reply src/secret_base.c]:
```u8 *roomdecor;
u8 *roomdecorpos;
u8 decidx;
u8 objid;```
They should also be camelCase. I added the comment to the bottom of the declarations and meant for it to include all of them.

**Pull request merged**